virtio-devices/src/mem.rs

```diff
@@ -180,6 +180,36 @@ struct VirtioMemConfig {
 // Safe because it only has data and has no implicit padding.
 unsafe impl ByteValued for VirtioMemConfig {}
 
+fn virtio_mem_config_resize(config: &mut VirtioMemConfig, size: u64) -> result::Result<(), Error> {
+    if config.requested_size == size {
+        return Err(Error::ResizeInval(format!(
+            "Virtio-mem resize {} is same with current config.requested_size",
+            size
+        )));
+    } else if size > config.region_size {
+        let region_size = config.region_size;
+        return Err(Error::ResizeInval(format!(
+            "Virtio-mem resize {} is bigger than config.region_size {}",
+            size, region_size
+        )));
+    } else if size % (config.block_size as u64) != 0 {
+        let block_size = config.block_size;
+        return Err(Error::ResizeInval(format!(
+            "Virtio-mem resize {} is not aligned with config.block_size {}",
+            size, block_size
+        )));
+    }
+
+    config.requested_size = size;
+    let tmp_size = cmp::min(
+        config.region_size,
+        config.requested_size + VIRTIO_MEM_USABLE_EXTENT,
+    );
+    config.usable_region_size = cmp::max(config.usable_region_size, tmp_size);
+
+    Ok(())
+}
```

Comment:
Could you please do the factorization of the existing code in a separate commit, so that we can easily identify the difference between factorization and code changes.